### DISCUSSION TITLE
Local plugin not being loaded

### DISCUSSION THREAD

**Discussion by jugglingcats** [gatsbyjs/gatsby#33269] [Help] [answered]
Hi, am trying to develop a simple local plugin in my gatsby project. The steps I've taken are:

1. Created new project using all defaults
```
npm init gatsby
```

2. Added `plugins/gatsby-plugin-simple` with `index.js` as follows (in desperate attempt to get something output!)
```
console.error("PLUGIN LOADED!!")
throw new Error("EXIT POINT 1")

exports.onPreInit = () => {
    console.error("PLUGIN PRE-INIT!!")
    throw new Error("EXIT POINT 2")
}
```

3. Added plugin to `gatsby-config.js`:
```
module.exports = {
    siteMetadata: {
        siteUrl: `https://www.yourdomain.tld`,
    },
    plugins: [
        "gatsby-plugin-simple"
    ]
}
```

Project structure:
![image](https://user-images.githubusercontent.com/526349/134162836-aced8e93-f9b1-42ac-9acc-cdb850c3aa2b.png)

Running `npm run develop` produces the following:
```
success open and validate gatsby-configs, load plugins - 0.191s
success onPreInit - 0.002s
success initialize cache - 0.074s
success copy gatsby files - 0.179s
success Compiling Gatsby Functions - 0.369s
```

If I deliberately misspell the plugin name in `gatsby-config.js` it gives an error, so I know it is resolving the plugin!

Output from `gatsby info`:
```
  System:
    OS: Windows 10 10.0.19043
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
  Binaries:
    Node: 14.17.3 - D:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - D:\Program Files\nodejs\yarn.CMD
    npm: 6.14.4 - D:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.9.6 - /c/Python39/python
  Browsers:
    Chrome: 93.0.4577.82
    Edge: Spartan (44.19041.1023.0), Chromium (93.0.961.52)
  npmPackages:
    gatsby: ^3.13.1 => 3.14.0 
  npmGlobalPackages:
    gatsby-cli: 3.0.0
```

Any help much appreciated, am at a bit of a loss.

**Comment by jugglingcats** [comment]


**Reply by hgb123** [reply]
_(moved to [this answer](https://github.com/gatsbyjs/gatsby/discussions/33269#discussioncomment-1391065))_

**Reply by jugglingcats** [reply]
Thank you that worked! There should be a mention in the docs, or even better Gatsby should use the `main` from `package.json`, imho.

I can't see how to mark your reply as the accepted answer...

**Reply by hgb123** [reply]
@jugglingcats I'll post that in a separate answer

**Comment by hgb123** [comment, marked as answer]
you should remove `index.js` and add `gatsby-node.js` instead, with the same content

```
// plugins/gatsby-plugin-simple/gatsby-node.js

console.error("PLUGIN LOADED!!")
throw new Error("EXIT POINT 1")

exports.onPreInit = () => {
    console.error("PLUGIN PRE-INIT!!")
    throw new Error("EXIT POINT 2")
}
```

and the `plugins/gatsby-plugin-simple/package.json` should be like
```
{
  "name": "gatsby-plugin-simple",
  ...
}
```
